### DISCUSSION TITLE
How to open popup witch startUrls ?

### DISCUSSION THREAD

**Discussion by ManUtopiK** [wxt-dev/wxt#1257] [Q&A] [answered]
I use the popup, but it's annoying to open the popup each time I reload the server and reopen chromium.
The url of my popup is `chrome-extension://gppniminfdpggjnapojieikjfnodlpnp/popup.html`. So if I open a page with this url, I can see my popup in a page. Hot reloading still working.
So I tried to open this page when chromium start with this config :
```js
runner: {
    chromiumArgs: ['--user-data-dir=./.wxt/chrome-data'],
    startUrls: ['chrome-extension://gppniminfdpggjnapojieikjfnodlpnp/popup.html'],
  }
```
Here I got a strange behavior. Chromium open with the url, but it's blocked (ERR_BLOCKED_BY_CLIENT). If I click on the button to actualize, nothing happen. If I place the cursor in the address bar and type Enter, it works !

Any idea to open the browser on the popup page ?

**Comment by ManUtopiK** [comment]
Also, can we got the extension hash `gppniminfdpggjnapojieikjfnodlpnp` by a variable in the runner config ?

**Comment by aklinker1** [comment, marked as answer]
ERR_BLOCKED_BY_CLIENT is what happens when a non-user action tries to open an extension URL, that's just how chrome works unfortunately.

As for the extension ID, you can make it consistent: https://developer.chrome.com/docs/extensions/reference/manifest/key#keep-consistent-id

To speed up opening your popup, I'd suggest adding a keyboard shortcut to open the popup during development. These are called "commands" in extensions: https://stackoverflow.com/questions/70795720/chrome-extension-open-popup-by-command#72646720
